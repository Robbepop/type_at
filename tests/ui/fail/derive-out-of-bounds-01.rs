use type_at::TypeAt;

#[derive(TypeAt)]
pub struct TupleStruct(i8, i16);

fn main() {
    let _: _  = <TupleStruct as TypeAt<2>>::Type::default();
}
