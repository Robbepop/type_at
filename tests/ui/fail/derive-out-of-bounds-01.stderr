error[E0277]: the trait bound `TupleStruct: TypeAt<2_usize>` is not satisfied
 --> tests/ui/fail/derive-out-of-bounds-01.rs:7:17
  |
7 |     let _: _  = <TupleStruct as TypeAt<2>>::Type::default();
  |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `TypeAt<2_usize>` is not implemented for `TupleStruct`
  |
  = help: the following implementations were found:
            <TupleStruct as TypeAt<0_usize>>
            <TupleStruct as TypeAt<1_usize>>
